packages/material-ui/src/MobileStepper/MobileStepper.test.js

```diff
@@ -40,92 +46,81 @@ describe('<MobileStepper />', () => {
+    expect(container.firstChild).to.have.class(classes.positionBottom);
   });
 
   it('should render with the top class if position prop is set to top', () => {
-    const wrapper = mount(<MobileStepper {...defaultProps} position="top" />);
-    expect(findOutermostIntrinsic(wrapper).hasClass(classes.positionTop)).to.equal(true);
+    const { container } = render(<MobileStepper {...defaultProps} position="top" />);
+    expect(container.firstChild).to.have.class(classes.positionTop);
   });
 
   it('should render two buttons', () => {
-    const wrapper = mount(<MobileStepper {...defaultProps} />);
-    expect(wrapper.find(Button)).to.have.lengthOf(2);
+    render(<MobileStepper {...defaultProps} />);
+    expect(screen.getAllByRole('button')).to.have.lengthOf(2);
   });
 
   it('should render the back button', () => {
-    const wrapper = mount(<MobileStepper {...defaultProps} />);
-    const backButton = wrapper.find('button[aria-label="back"]');
-    expect(backButton.exists()).to.equal(true);
-    expect(backButton.find('svg[data-testid="KeyboardArrowLeftIcon"]')).to.have.lengthOf(1);
+    const { container } = render(<MobileStepper {...defaultProps} />);
+    const backButton = container.firstChild.querySelector('button[aria-label="back"]');
+    expect(backButton).to.not.equal(null);
+    expect(backButton.querySelector('svg[data-testid="KeyboardArrowLeftIcon"]')).to.not.equal(null);
   });
 
   it('should render next button', () => {
-    const wrapper = mount(<MobileStepper {...defaultProps} />);
-    const nextButton = wrapper.find('button[aria-label="next"]');
-    expect(nextButton.exists()).to.equal(true);
-    expect(nextButton.find('svg[data-testid="KeyboardArrowRightIcon"]')).to.have.lengthOf(1);
+    const { container } = render(<MobileStepper {...defaultProps} />);
+    const nextButton = container.firstChild.querySelector('button[aria-label="next"]');
+    expect(nextButton).to.not.equal(null);
+    expect(nextButton.querySelector('svg[data-testid="KeyboardArrowRightIcon"]')).to.not.equal(
+      null,
+    );
   });
 
   it('should render two buttons and text displaying progress when supplied with variant text', () => {
-    const wrapper = mount(
+    const { container } = render(
       <MobileStepper {...defaultProps} variant="text" activeStep={1} steps={3} />,
     );
-    expect(findOutermostIntrinsic(wrapper).instance().textContent).to.equal('Back2 / 3Next');
+    expect(container.firstChild.textContent).to.equal('Back2 / 3Next');
   });
 
   it('should render dots when supplied with variant dots', () => {
-    const wrapper = mount(<MobileStepper {...defaultProps} variant="dots" />);
-    const outermost = findOutermostIntrinsic(wrapper);
-    expect(outermost.children()).to.have.lengthOf(3);
-    expect(outermost.childAt(1).hasClass(classes.dots)).to.equal(true);
+    const { container } = render(<MobileStepper {...defaultProps} variant="dots" />);
+    expect(container.firstChild.children).to.have.lengthOf(3);
```

Comment:
Use a css selector instead of diving too deep in the DOM structure. Apply to the other cases.

Reply:
You mean something like this
```javascript
expect(container.querySelectorAll(`.${classes.root} > *`)).to.have.lengthOf(2);
```
@oliviertassinari

Reply:
```suggestion
    expect(container.querySelectorAll(`.${classes.dot}`)).to.have.lengthOf(1);
```

Reply:
The step is prop in defaultProps is 2, and that is tested in this test case "should render a dot for each step when using dots variant"

Reply:
Just updated with more fixes :)